Userland/Libraries/LibWeb/Infra/Strings.cpp

```diff
@@ -65,4 +66,129 @@ bool is_code_unit_prefix(StringView potential_prefix, StringView input)
     }
 }
 
+static constexpr Array base64_alphabet = {
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H',
+    'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+    'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+    'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n',
+    'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+    'w', 'x', 'y', 'z', '0', '1', '2', '3',
+    '4', '5', '6', '7', '8', '9', '+', '/'
+};
+
+static consteval auto make_base64_lookup_table()
+{
+    Array<i16, 256> table;
+    table.fill(-1);
+    for (size_t i = 0; i < base64_alphabet.size(); ++i) {
+        table[base64_alphabet[i]] = static_cast<i16>(i);
+    }
+    return table;
+}
+
+static constexpr auto base64_alphabet_lookup_table = make_base64_lookup_table();
+
+// https://infra.spec.whatwg.org/#forgiving-base64
+ErrorOr<ByteBuffer> decode_forgiving_base64(StringView input)
```

Comment:
Strings.cpp is section "4.6 Strings" of the Infra spec. Please put this in its own Base64.cpp file.